### PR TITLE
Remove unhelpful benchmarks

### PR DESCRIPTION
**Pull request by iNikem**
Part of #4483

I reviewed benchmarks in `benchmark` module and the majority of them did not make sense to me. I moved a couple of useful ones to a module that contains classes they benchmark and removed the whole remaining module.

### CODE REVIEW

File: benchmark/src/jmh/java/io/opentelemetry/benchmark/IgnoredTypesMatcherBenchmark.java [hunk 1]
```diff
@@ -1,52 +0,0 @@
...
-import org.openjdk.jmh.annotations.Mode;
-import org.openjdk.jmh.annotations.OutputTimeUnit;
-import org.openjdk.jmh.annotations.Warmup;
-
-@Fork(1)
-@Warmup(iterations = 3, time = 1)
-@Measurement(iterations = 10, time = 1)
-@OutputTimeUnit(TimeUnit.MICROSECONDS)
-@BenchmarkMode(Mode.AverageTime)
-public class IgnoredTypesMatcherBenchmark {
```
Comment (trask) [inline benchmark/src/jmh/java/io/opentelemetry/benchmark/IgnoredTypesMatcherBenchmark.java:28]: this one seems worth keeping (cc @mateuszrzeszutek)

**Review by trask — APPROVED** [review]

File: benchmark/src/jmh/java/io/opentelemetry/benchmark/IgnoredTypesMatcherBenchmark.java (hunk 1 above)
Comment (iNikem) [reply benchmark/src/jmh/java/io/opentelemetry/benchmark/IgnoredTypesMatcherBenchmark.java:28]: Then I will ask some clarification about what it does :) What is the difference between `springType` and `appType`?

Comment (mateuszrzeszutek) [reply benchmark/src/jmh/java/io/opentelemetry/benchmark/IgnoredTypesMatcherBenchmark.java:28]: The length of the matched prefix -- `com.example.myapp.Main` is dropped after `com.`, `org.springframework.test.SomeClass` after `org.springframework.`. Maybe there's a better example now, but when I wrote this benchmark and compared against the previous implementation it had to be spring - there were additional ifs, nesting and string comparisons for spring classes.

Comment (iNikem) [reply benchmark/src/jmh/java/io/opentelemetry/benchmark/IgnoredTypesMatcherBenchmark.java]: I restored this benchmark

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**